samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -220,10 +223,14 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
     * @return           the map of properties for the stream
     */
   private def getSystemStreamProperties(systemName: String, streamName: String) = {
-    if (systemName == null || streamName == null) {
+    if (systemName == null) {
```

Comment:
I don't see the need for the distinction. 

Legacy: "systems.sysName.streams.streamName" accessed by getSystemStreamProperties
New: "streams.streamId" accessed by getAllStreamProperties

They're different enough IMO. And hopefully soon the legacy approach will be completely removed, and this method wont be needed.
If you feel strongly about it, I can change it. Just didn't see the need.